claim.go

```diff
@@ -101,15 +101,28 @@ func NewSchemaHashFromHex(s string) (SchemaHash, error) {
 		return SchemaHash{}, err
 	}
 
-	if len(schemaEncodedBytes) != len(schemaHash) {
+	if len(schemaEncodedBytes) > len(schemaHash) {
 		return SchemaHash{}, fmt.Errorf("invalid schema hash length: %d",
 			len(schemaEncodedBytes))
 	}
-	copy(schemaHash[:], schemaEncodedBytes)
+	copy(schemaHash[len(schemaHash)-len(schemaEncodedBytes):], schemaEncodedBytes)
 
 	return schemaHash, nil
 }
 
+// NewSchemaHashFromInt creates new SchemaHash from big.Int
+func NewSchemaHashFromInt(i *big.Int) SchemaHash {
+	var schemaHash SchemaHash
+	i.FillBytes(schemaHash[:])
```

Comment:
Other types use little-endian to serialize big.Int to []byte. I would prefer to use `bytesToInt` & `intToBytes` functions to construct `SchemaHash` from int or represent `SchemaHash` as `[]byte` in `SchmaHash.BigInt()` method.

Reply:
Changed